wazimap/urls.py

```diff
@@ -1,235 +1,209 @@
 from django.conf import settings
 from django.conf.urls import url, include
 from django.contrib import admin
-from django.core.urlresolvers import reverse_lazy
+from django.urls import reverse_lazy, path, re_path
 from django.http import HttpResponse
 from django.views.decorators.cache import cache_page
 from django.views.generic.base import RedirectView, TemplateView
 
 from census.views import HealthcheckView, DataView, ExampleView
 
-from wazimap.views import (HomepageView, GeographyDetailView, GeographyJsonView, PlaceSearchJson,
-                           LocateView, DataAPIView, TableAPIView, AboutView, HelpView, GeographyCompareView,
-                           GeoAPIView, TableDetailView)
+from wazimap.views import (
+    HomepageView,
+    GeographyDetailView,
+    GeographyJsonView,
+    PlaceSearchJson,
+    LocateView,
+    DataAPIView,
+    TableAPIView,
+    AboutView,
+    HelpView,
+    GeographyCompareView,
+    GeoAPIView,
+    TableDetailView,
+)
 
 
-#admin.autodiscover()
-handler500 = 'census.views.server_error'
+# admin.autodiscover()
+handler500 = "census.views.server_error"
 
-STANDARD_CACHE_TIME = settings.WAZIMAP['cache_secs']
-EMBED_CACHE_TIME = settings.WAZIMAP.get('embed_cache_secs', STANDARD_CACHE_TIME)
+STANDARD_CACHE_TIME = settings.WAZIMAP["cache_secs"]
+EMBED_CACHE_TIME = settings.WAZIMAP.get("embed_cache_secs", STANDARD_CACHE_TIME)
 
 
 urlpatterns = [
-    url(r'^admin/', include(admin.site.urls)),
-
-    url(
-        regex   = '^$',
-        view    = cache_page(STANDARD_CACHE_TIME)(HomepageView.as_view()),
-        kwargs  = {},
-        name    = 'homepage',
-    ),
-
-    url(
-        regex   = '^about$',
-        view    = cache_page(STANDARD_CACHE_TIME)(AboutView.as_view()),
-        kwargs  = {},
-        name    = 'about',
-    ),
-    url(
-        regex   = '^help$',
-        view    = cache_page(STANDARD_CACHE_TIME)(HelpView.as_view()),
-        kwargs  = {},
-        name    = 'help',
+    path("admin/", admin.site.urls),
+    path("", cache_page(STANDARD_CACHE_TIME)(HomepageView.as_view()), name="homepage"),
+    path("about", cache_page(STANDARD_CACHE_TIME)(AboutView.as_view()), name="about"),
+    path("help", cache_page(STANDARD_CACHE_TIME)(HelpView.as_view()), name="help"),
+    # e.g. /profiles/province/GT/slug
+    path(
+        "profiles/<geo_level>/<geo_code>/<slug>",
```

Comment:
Why have the profiles path been changed?

Reply:
Which Django to the recommended way of structuring your URLs is with the new path function and separating with / instead of -. Also breaking down the url with geo_level/geo_code makes things a lot easier to understand, instead of having to split the geo_id everywhere.